BeerProject/src/test/java/wawer/kamil/beerproject/controllers/UserControllerTest.java

```diff
@@ -1,125 +1,248 @@
 import wawer.kamil.beerproject.exceptions.NoContentException;
+import wawer.kamil.beerproject.exceptions.UsernameAlreadyExistsException;
+import wawer.kamil.beerproject.model.User;
 import wawer.kamil.beerproject.service.UserService;
 
 import java.util.List;
 
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.mockito.Mockito.when;
+import static wawer.kamil.beerproject.controllers.UserControllerTestHelper.*;
 
-@RunWith(MockitoJUnitRunner.class)
+@ExtendWith(MockitoExtension.class)
 public class UserControllerTest {
 
     @Mock
-    UserService service;
-
-    @Mock
-    Pageable pageable;
-
-    @Mock
-    Page<User> page;
-
-    @Mock
-    List<User> userList;
-
-    @Mock
-    ModelMapper mapper;
-
-    @Mock
-    User user;
-
-    @Mock
-    UserDTO userDTO;
+    UserService userService;
 
     @InjectMocks
     UserController controller;
 
-    private final static Long ID = 1L;
+    private UserResponse userResponse;
+    private Page<UserResponse> userResponsePage;
+    private List<UserResponse> userResponseList;
+    private Pageable pageable;
+    private UserRequest userRequest;
+
+    private final static Long USER_ID = 1L;
+    private final static int PAGE_SIZE = 1;
+    private final static int PAGE_PAGE = 1;
+
+    @BeforeEach
+    void setUp() {
+        this.userResponse = createUserResponse();
+        this.userResponsePage = createPage();
+        this.userResponseList = createListOfUsers();
+        this.pageable = createPageable(PAGE_PAGE, PAGE_SIZE);
+        this.userRequest = createUserRequest();
+    }
 
     @Test
-    public void should_return_response_entity_which_equals_to_controller_response_entity_user_page() {
-        when(service.findAllUsersPage(pageable)).thenReturn(page);
-        assertEquals(ResponseEntity.ok().body(page), controller.findAllUsersPage(pageable));
+    @DisplayName("Test - should return response entity with generated user")
+    public void should_return_response_entity_with_generated_user() {
+        //when
+        ResponseEntity<User> userResponseEntity = controller.generateUser();
+
+        //than
+        assertThat(userResponseEntity).isNotNull();
+        assertThat(userResponseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(userResponseEntity.getBody()).isNull();
     }
 
     @Test
-    public void should_return_status_code_which_equals_to_controller_status_code_user_page() throws NoContentException {
-        when(service.findAllUsersPage(pageable)).thenReturn(page);
-        assertEquals(HttpStatus.OK, controller.findAllUsersPage(pageable).getStatusCode());
+    @DisplayName("Test - should return response entity with page body")
+    public void should_return_response_entity_with_page_body() {
+        //given
+        when(userService.findAllUsersPage(pageable)).thenReturn(userResponsePage);
+
```

Comment:
to jak to jest w sekcji // given to może zaminieć na given(userService..) albo obecne when umieścić w sekcji //when

Reply:
w sekcji given umieszczamy wszelkie "konfiguracje" i wywołania metod, które robią nam setup do danego testu. W sekcji when wywołujemy konkretną metode do testu. 
A jeśli chodzi o konwencje to nazywałem to tak jak w kursie testów jednostkowych od gościa z nullpointerexception. On tak robił to ja za nim powtórzyłem

Reply:
ok, to zostawmy tak. Musze sobie ten kurs też przerobić.